goaccess: update to 1.4

alm0ndiga opened the pull request:
Hi all.
I am using OpenIndiana for some time and i would like contribute. This should be my first pull request. I am ready to do any amendments.
Thanks.

alm0ndiga committed:
goaccess: update to 1.4

Toasterson commented on components/web/goaccess/goaccess.p5m:
```diff
@@ -22,14 +23,17 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
-file path=etc/goaccess.conf preserve=true
```
This got deleted during the update of the manifest. Please leave existing manual additions in when updating.

Toasterson replied:
Please also make sure that config file locations don't change unless absolutely necessary. Existing users will need to be notified about the update otherwise.

alm0ndiga replied:
Thank you Toasterson, it's fixed, now the previous config file is there with preserve=true.

Toasterson commented on components/web/goaccess/goaccess.p5m:
```diff
@@ -22,14 +23,17 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
-file path=etc/goaccess.conf preserve=true
+dir path=etc/goaccess
```
Directory actions are only required, if no other files below are owned by this package. The directories will be created automatically if needed. Please remove.

alm0ndiga replied:
Thanks, it's removed now.

Toasterson commented on components/web/goaccess/manifests/sample-manifest.p5m:
```diff
@@ -22,14 +22,15 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 
-file path=etc/goaccess.conf
+file path=etc/goaccess/browsers.list
+file path=etc/goaccess/goaccess.conf
```
Please include preserve=true to config files the user should be able to update, otherwise a "pkg fix" call to restore system state will overwrite user changes.

alm0ndiga replied:
Thanks Toasterson, now it's included preserve=true in config files

Toasterson reviewed (changes requested):
See my comments otherwise LGTM

AndWac commented on components/web/goaccess/goaccess.p5m:
```diff
@@ -11,6 +11,7 @@
 #
 # Copyright 2016 Adam Stevko
+# Copyright 2016 Pablo Sarria
 #
```
2020 :)

alm0ndiga replied:
Thanks AndWac, it's fixed now, i absolutely forget the year !!! :D

alm0ndiga committed:
corrected Makefile

Add BUILD_BITS, replace includes per common.mk, delete make targets (build, install, test)

alm0ndiga committed:
corrected goaccess.p5m

Leave previous file path for goaccess.conf with preserve=true, delete dir path, add preserve=true to config files browsers.list and new goaccess.conf and finally, date corrected :)

alm0ndiga committed:
update sample-manifest.p5m

Added previous config file (1.2 version) with preserve=true, also added preserve=true to browser.list and goaccess.conf in its new location

AndWac commented on components/web/goaccess/manifests/sample-manifest.p5m:
```diff
@@ -22,14 +22,16 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+file path=etc/goaccess.conf preserve=true
+file path=etc/goaccess/browsers.list preserve=true
+file path=etc/goaccess/goaccess.conf preserve=true
+file path=etc/goaccess/podcast.list
```
Never touch the sample-manifest. Only generate it and add to your PR if it has changes other than the year.
With this you (or any other maintainer) can check whether something has changed since the last release.

AndWac commented on components/web/goaccess/Makefile:
```diff
@@ -11,43 +11,35 @@
 	$(CLONEY) $(SOURCE_DIR) $(@D)
 
 CONFIGURE_BINDIR.64= $(CONFIGURE_BINDIR.32)
 
```
You need to remove this line. At least on my build system it is responsible for the $(MACH32) in the file action:
file path=usr/bin/$(MACH32)/goaccess

alm0ndiga commented:
Great! Thanks a lot AndWac and Toasterson for make it possible :)

Toasterson commented:
You only got one of two approvals. Look in the thread above for @AndWac comments. Those you two things you will still need to fix.

alm0ndiga committed:
delete configure_bindir.32

Delete this line in order to avoid MACH32 in sample-manifest.p5m

alm0ndiga commented:
Thanks Toasterson, I've deleted right now that line in Makefile in order to avoid MACH32 line in sample-manifest.p5m.

alm0ndiga committed:
updated sample-manifest.p5m to original

updated sample-manifest.p5m to original as generated by 'gmake sample-manifest'

alm0ndiga commented:
Here, I've updated sample-manifest.p5m to original as generated by 'gmake sample-manifest'

AndWac commented on components/web/goaccess/goaccess.p5m:
```diff
@@ -23,13 +24,15 @@ set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
 license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
 file path=etc/goaccess.conf preserve=true
```
Did you run "gmake publish"? On my machine it fails:
Couldn't find 'etc/goaccess.conf' in any of the specified search directories:
	/oibuild/oi-userland/components/web/goaccess
	/oibuild/oi-userland/components/web/goaccess/build
	/oibuild/oi-userland/components/web/goaccess/build/prototype/i386
	/oibuild/oi-userland/components/web/goaccess/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/goaccess/goaccess-1.4
gmake: *** [/oibuild/oi-userland/make-rules/ips.mk:434: /oibuild/oi-userland/components/web/goaccess/build/manifest-i386-goaccess.depend] Error 1

AndWac replied:
So, your idea with saving the old goaccess.conf file is fine but doesn't work as it is now. I propose to remove the line and warn possible users over the mailing list resp. release notes.
It is probably possible to do what you intended to do by means of bypass-generate  but I wouldn't do that here.

Toasterson reviewed (changes requested):
Hi, @alm0ndiga

Sorry, I just looked into the goaccess configs, and they have changed sufficiently that we need to remove the preserve=true. Also `goaccess.conf` has moved from `etc` to `etc/goaccess` so we should only deliver the new config file this update.

alm0ndiga committed:
update goaccess.p5m deleting old config file

Deleted old config file with preserve=true

alm0ndiga commented:
I've modified goaccess.p5m deleting old config file

AndWac commented:
Thank you for your valuable work and congratulations to your first contribution!
